searchcore/src/vespa/searchcore/proton/attribute/attribute_directory.h

```diff
@@ -50,6 +49,9 @@ class AttributeDirectory
                        const vespalib::string &name);
     ~AttributeDirectory();
 
+    const vespalib::string & getAttrName() const { return _name; }
+    vespalib::string getSnapshotDir(SerialNum serialNum);
```

Comment:
getSnapshotDir() should only be callable via nested Writer class (to ensure it isn't removed by another thread while we're using it), or via getAttributeFileName() (when loading attribute at startup).

Reply:
I moved it back to private scope. It was not needed in this PR anyway.